cli/lsp/jsr_resolver.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use crate::args::deno_registry_url;
+use deno_cache_dir::HttpCache;
+use deno_core::parking_lot::Mutex;
+use deno_core::serde_json;
+use deno_core::serde_json::json;
+use deno_core::ModuleSpecifier;
+use deno_graph::packages::JsrPackageVersionInfo;
+use deno_lockfile::Lockfile;
+use deno_semver::jsr::JsrPackageReqReference;
+use deno_semver::package::PackageNv;
+use deno_semver::package::PackageReq;
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Default)]
+pub struct JsrResolver {
+  nv_by_req: HashMap<PackageReq, PackageNv>,
+  /// The `module_graph` field of the version infos should be forcibly absent.
+  /// It can be large and we don't want to store it.
+  info_by_nv: HashMap<PackageNv, JsrPackageVersionInfo>,
+}
+
+// TODO(nayeemrmn): Derive `Debug` on `PackageReq` etc..
```

Comment:
PackageReq and PackageNv does derive debug: https://github.com/denoland/deno_semver/blob/0cff13bdb24318e7c7423e22360b2e4e8bbdd585/src/package.rs#L148-L149

Seems like it's missing on `JsrPackageVersionInfo`. Opened https://github.com/denoland/deno_graph/pull/376